opnode/rollup/sync/start_test.go

```diff
@@ -90,224 +99,217 @@ type syncStartTestCase struct {
+			ExpectedRefL2:      'A',
+			ExpectedErr:        nil,
 		},
 		{
-			Name:              "Orphan block",
-			OffsetL2:          0,
-			EngineL1:          "abcd",
-			EngineL2:          "ABCD",
-			ActualL1:          "abcx",
-			GenesisL1:         'a',
-			GenesisL2:         'A',
-			ExpectedNextRefL1: 'x',
-			ExpectedRefL2:     'C',
-			ExpectedErr:       nil,
+			Name:               "Orphan block",
+			OffsetL2:           0,
+			EngineL1:           "abcd",
+			EngineL2:           "ABCD",
+			ActualL1:           "abcx",
+			GenesisL1:          'a',
+			GenesisL2:          'A',
+			ExpectedNextRefsL1: "x",
+			ExpectedRefL2:      'C',
+			ExpectedErr:        nil,
 		},
 		{
-			Name:              "L2 chain ahead",
-			OffsetL2:          0,
-			EngineL1:          "abcdef",
-			EngineL2:          "ABCDEF",
-			ActualL1:          "abc",
-			GenesisL1:         'a',
-			GenesisL2:         'A',
-			ExpectedNextRefL1: 0,
-			ExpectedRefL2:     0,
-			ExpectedErr:       ethereum.NotFound,
+			Name:               "L2 chain ahead",
+			OffsetL2:           0,
+			EngineL1:           "abcdef",
+			EngineL2:           "ABCDEF",
+			ActualL1:           "abc",
+			GenesisL1:          'a',
+			GenesisL2:          'A',
+			ExpectedNextRefsL1: "",
+			ExpectedRefL2:      'C',
+			ExpectedErr:        nil,
 		},
 		{
-			Name:              "L2 chain ahead reorg",
-			OffsetL2:          0,
-			EngineL1:          "abcdef",
-			EngineL2:          "ABCDEF",
-			ActualL1:          "abcx",
-			GenesisL1:         'a',
-			GenesisL2:         'A',
-			ExpectedNextRefL1: 'x',
-			ExpectedRefL2:     'C',
-			ExpectedErr:       nil,
+			Name:               "L2 chain ahead reorg",
+			OffsetL2:           0,
+			EngineL1:           "abcdef",
+			EngineL2:           "ABCDEF",
+			ActualL1:           "abcx",
+			GenesisL1:          'a',
+			GenesisL2:          'A',
+			ExpectedNextRefsL1: "x",
+			ExpectedRefL2:      'C',
+			ExpectedErr:        nil,
 		},
 		{
-			Name:              "unexpected L1 chain",
-			OffsetL2:          0,
-			EngineL1:          "abcdef",
-			EngineL2:          "ABCDEF",
-			ActualL1:          "xyz",
-			GenesisL1:         'a',
-			GenesisL2:         'A',
-			ExpectedNextRefL1: 0,
-			ExpectedRefL2:     0,
-			ExpectedErr:       WrongChainErr,
+			Name:               "unexpected L1 chain",
+			OffsetL2:           0,
+			EngineL1:           "abcdef",
+			EngineL2:           "ABCDEF",
+			ActualL1:           "xyz",
+			GenesisL1:          'a',
+			GenesisL2:          'A',
+			ExpectedNextRefsL1: "",
+			ExpectedRefL2:      0,
+			ExpectedErr:        WrongChainErr,
 		},
 		{
-			Name:              "unexpected L2 chain",
-			OffsetL2:          0,
-			EngineL1:          "abcdef",
-			EngineL2:          "ABCDEF",
-			ActualL1:          "xyz",
-			GenesisL1:         'a',
-			GenesisL2:         'X',
-			ExpectedNextRefL1: 0,
-			ExpectedRefL2:     0,
-			ExpectedErr:       WrongChainErr,
+			Name:               "unexpected L2 chain",
```

Comment:
@protolambda , this is the same as the above test. Was there something you were thinking here or should we cut them down to 1?

Reply:
They are not the same, the genesis configuration is different. You can have the wrong L1 chain at genesis, or the wrong L2 chain at genesis. E.g. due to connecting to the wrong RPC endpoint. In both cases sync cannot proceed and should abort.

Reply:
Oh I see that now. Will leave in.